firebase-database/src/androidTest/java/com/google/firebase/database/QueryTest.java

```diff
@@ -4548,6 +4548,57 @@ private static FirebaseApp appForDatabaseUrl(String url, String name) {
         name);
   }
 
+  @Test
+  public void testGetPropagatesEventsToListeners() // TODO(mtewani): Convert into smoke test
+      throws DatabaseException, InterruptedException, ExecutionException {
+    /**
+     * There's a situation where if you have a pending get() and then an onValue() at the same
+     * location and get resolves before onValue does, onValue gets no listeners
+     */
+    int counter = 0;
+    boolean pendingGet = false;
+    while (counter++ < 5 || !pendingGet) {
+      FirebaseApp writeApp =
+          appForDatabaseUrl(IntegrationTestValues.getDatabaseUrl(), UUID.randomUUID().toString());
+      // To ensure that we don't read the cached results, we need a separate app.
+      FirebaseApp readApp =
+          appForDatabaseUrl(IntegrationTestValues.getDatabaseUrl(), UUID.randomUUID().toString());
+      FirebaseDatabase readDb = FirebaseDatabase.getInstance(readApp);
+      FirebaseDatabase writeDb = FirebaseDatabase.getInstance(writeApp);
+      DatabaseReference writeNode = writeDb.getReference().push();
+      DatabaseReference readNode = IntegrationTestHelpers.translateReference(writeNode, readDb);
+      try {
+        Map<String, Object> toWrite =
+            new MapBuilder().put("foo1", "child1").put("foo2", "child2").build();
+        await(writeNode.setValue(toWrite));
+        Task<DataSnapshot> getTask = readNode.get();
+        AtomicBoolean receivedGetValue = new AtomicBoolean();
+        getTask.addOnCompleteListener(
```

Comment:
Couldn't you ensure that `getTask` is pending by taking the database offline, calling `get()`, register the `ValueEventRegistration`, and then going back online?